src/PiWeb.Import.Sdk/Modules/ImportFormat/ImportGroupExtensions.cs

```diff
@@ -27,11 +27,12 @@ public static class ImportGroupExtensions
     /// Creates an <see cref="ImportGroup"/> that is complete and can be imported at once.
     /// </summary>
     /// <param name="associatedFile">The file to be imported.</param>
-    public static ImportGroup CreateCompleteImportGroup(this IFileSource associatedFile)
+    /// <param name="groupState">The state of the resulting <see cref="ImportGroup"/>.</param>
+    public static ImportGroup CompleteGroup(this IFileSource associatedFile, State groupState)
```

Comment:
Shouldn't this be also called `CreateImportGroup`? The `Complete` seems to be misleading if the user can pass any `groupState` here.

Reply:
We should ensure that the developer must pass the `primaryInputSource` to that group.

What do you think about changing `IImportFormat` interface slightly 
from `ImportGroup? BuildGroup(IFileSource primaryFileSource, IGroupContext context)` 
to `ImportGroup? BuildGroup(IPrimaryFileSource primaryFileSource, IGroupContext context)` 
and also add an interface like `IPrimaryFileSource : IFileSource`?

That way we could enforce the `primaryInputSource` to be part of that group.

Reply:
Good idea. I introduced `IPrimaryFileSource `and altered the code accordingly.